src/code/z_msgevent.c

```diff
@@ -1,105 +1,1229 @@
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_msgevent/func_8010BF58.s")
+    Rupees_ChangeBy(rupees);
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Pause the message system
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_Pause(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    play->msgCtx.msgMode = 0x44;
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Unsets ACTOR_FLAG_10000 for the actor executing the script
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_UnsetAutotalk(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    actor->flags &= ~ACTOR_FLAG_10000;
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Sets player focus & talk actor to the child of the actor executing the script
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_FocusToChild(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    if (actor->child != NULL) {
+        Actor_ChangeFocus(actor, play, actor->child);
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Sets player focus & talk actor to the actor executing the script
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ * Command size: 1
+ */
+s32 MsgEvent_FocusToSelf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    if (actor->child != NULL) {
+        Actor_ChangeFocus(actor->child, play, actor);
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Unconditional branch by `skip`
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(s16) skip
+ * Command size: 3
+ */
+s32 MsgEvent_Jump(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback, s32* endScript) {
+    u8* script = *scriptPtr;
+    s16 skip = MSCRIPT_GET_16(script, 1);
+
+    *scriptPtr += skip;
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Branch forwards if the specified quest item is obtained
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) questItem
+ *  3:(s16) skip
+ * Command size: 5
+ */
+s32 MsgEvent_BranchOnQuestItem(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback,
+                               s32* endScript) {
+    u8* script = *scriptPtr;
+    u16 questItem = MSCRIPT_GET_16(script, 1);
+    s16 skip = MSCRIPT_GET_16(script, 3);
+
+    if (CHECK_QUEST_ITEM(questItem)) {
+        *scriptPtr += skip;
+    }
+    return MSCRIPT_CONTINUE;
+}
+
+/**
+ * Branch forwards if event_inf flag is set
+ *
+ * Command structure:
+ *  0:(u8)  cmd
+ *  1:(u16) event_inf flag
+ *  3:(s16) skip
+ * Command size: 5
+ */
+s32 MsgEvent_BranchOnEventInf(Actor* actor, PlayState* play, u8** scriptPtr, MsgEventCallback callback,
+                              s32* endScript) {
+    u8* script = *scriptPtr;
+    s16 skip = MSCRIPT_GET_16(script, 3);
+
+    if (MSCRIPT_GET_8(script, 2) & gSaveContext.eventInf[MSCRIPT_GET_8(script, 1)]) {
```

Comment:
I suppose there's no way to use the `CHECK_EVENTINF` macro here, right?

Reply:
I don't think so. The main difference here is how the message script stores the eventinf vs how we do in the defines. The script stores it as index and the raw flag, while for the define we store it as index and the flag position. So get the flag the macro expects we would need to put it back together with something like `(MSCRIPT_GET_8(script, 1) << 4) | BIT_FLAG_TO_SHIFT(MSCRIPT_GET_8(script, 2))` which doesn't match. Note this is fine for weekevent regs as we store them based on index and raw flag as well.

Reply:
Oh, I see.

In the future we could consider changing how the evenInf macro works then. Thanks for looking at this